### ISSUE TITLE
[Suggestion] Add a Fieldset element

### ISSUE THREAD

**Issue by monkeypox8** [Kajabi/sage-lib#1007]
## Description
Groups of related form elements (checkboxes and radio buttons in particular) stand to benefit from the addition of a Sage `fieldset` (and `legend`).

- Using a `fieldset` will clearly define the relationship between controls to screen readers
- Developers will gain the ability to enable and disable entire sets of controls at once
- Testing can be simplified by allowing grouped form controls to be scoped under a single `test_id`

Discussion points/considerations
- Styling `fieldset` styling: remove border and padding as defaults?
- `legend` should be *required* but provide an option to visually hide

## Additional Information
- [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset)
- https://accessibility.blog.gov.uk/2016/07/22/using-the-fieldset-and-legend-elements/
